.github/workflows/performance-checks.yml

```diff
@@ -0,0 +1,39 @@
+name: performance checks
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - performance/**
```

Comment:
Just want to double-check whether this should be scoped to `performance/samples/**`.